lib/clients/GcpClient.py

```diff
@@ -0,0 +1,553 @@
+        log_prefix = '[VOLUME] [DELETE]'
+
+        try:
+            disk_deletion_operation = self.compute_client.disks().delete(
+                project=self.project_id, zone=self.availability_zone, disk=volume_id).execute()
+
+            self._wait('Waiting for disk {} to be deleted...'.format(volume_id),
+                       (lambda operation_id, zonal_operation: self.get_operation_status(
+                           operation_id, zonal_operation) == 'DONE'),
+                       None,
+                       disk_deletion_operation['name'], True)
+
+            volume_exists = self.volume_exists(volume_id)
+
+            # Check if volume exists, if not then it is successfully deleted, else raise exception
+            if not volume_exists:
+                self._remove_volume(volume_id)
+                self.logger.info(
+                    '{} SUCCESS: volume-id={}'.format(
+                        log_prefix, volume_id))
+                return True
+            else:
+                message = '{} ERROR: volume-id={}, volume still exists'.format(
+                    log_prefix, volume_id)
+                self.logger.error(message)
+                raise Exception(message)
+        except Exception as error:
+            message = '{} ERROR: volume-id={}\n{}'.format(
+                log_prefix, volume_id, error)
+            self.logger.error(message)
+            raise Exception(message)
+
+    def _create_attachment(self, volume_id, instance_id):
+        log_prefix = '[ATTACHMENT] [CREATE]'
+        attachment = None
+
+        try:
+            volume = self.compute_client.disks().get(
+                project=self.project_id, zone=self.availability_zone, disk=volume_id).execute()
+            attached_disk_body = {
+                'source': volume['selfLink'],
+                'deviceName': volume_id
+                # type: the default is PERSISTENT
+            }
+
+            disk_attach_operation = self.compute_client.instances().attachDisk(
+                project=self.project_id, zone=self.availability_zone, instance=instance_id, body=attached_disk_body).execute()
+
+            self._wait('Waiting for attachment of volume {} to get ready...'.format(volume_id),
+                       (lambda operation_id, zonal_operation: self.get_operation_status(
+                           operation_id, zonal_operation) == 'DONE'),
+                       None,
+                       disk_attach_operation['name'], True)
+
+            # Here volume_id is the device name.
+            # Raise exception if device returned in None,
+            # as it might mean that disk was not attached properly
+            device = self._find_volume_device(volume_id)
+            if device is not None:
+                self.logger.info(
+                    'Attached volume-id={}, device={}'.format(volume_id, device))
+                self._add_volume_device(volume_id, device)
+            else:
+                raise Exception()
```

Comment:
Put some logs/ messages inside the Exception()

Reply:
Sure, will add.